src/Extractor/NodeJs/lib/Extractor.js

```diff
@@ -67,6 +67,11 @@ class Extractor {
   }
 
   async processPage(page, pageIndex, resolve) {
+    if (!page.resources) {
```

Comment:
Pri testovani tvojho PR som narazil na jednu chybu, ktora tam uz bola ... 

Ak nie je nacitany ziaden vysledok, tak `page.resources` je `[]` ... no pokial je v query nejaky key, ktory neexistuje v ziadnom dokumente (cosmosdb nema pevnu schemu) ... tak `page.resources` je `undefined`. Fixujem to a pridal som test ...